libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -0,0 +1,346 @@
+
+            return recognizerResult;
+        }
+
+        private void AddResultAsIntent(RecognizerResult recognizerResult, Result result)
+        {
+            recognizerResult.Intents.Add(result.Label.Name, new IntentScore()
+            {
+                Score = result.Score
+            });
+        }
+
+        private async Task<RecognizerResult> RecognizeEntitiesAsync(DialogContext dialogContext, Schema.Activity activity, RecognizerResult recognizerResult)
+        {
+            var text = activity.Text ?? string.Empty;
+            var entityPool = new List<Entity>();
+            if (EntityRecognizers != null)
+            {
+                // add entities from regexrecgonizer to the entities pool
+                var textEntity = new TextEntity(text);
+                textEntity.Properties["start"] = 0;
+                textEntity.Properties["end"] = text.Length;
+                textEntity.Properties["score"] = 1.0;
+
+                entityPool.Add(textEntity);
+
+                // process entities using EntityRecognizerSet
+                var entitySet = new EntityRecognizerSet(EntityRecognizers);
+                var newEntities = await entitySet.RecognizeEntitiesAsync(dialogContext, activity, entityPool).ConfigureAwait(false);
+                if (newEntities.Any())
+                {
+                    entityPool.AddRange(newEntities);
+                }
+
+                entityPool.Remove(textEntity);
+            }
+
+            // map entityPool of Entity objects => RecognizerResult entity format
+            recognizerResult.Entities = new JObject();
+
+            foreach (var entityResult in entityPool)
+            {
+                // add value
+                JToken values;
+                if (!recognizerResult.Entities.TryGetValue(entityResult.Type, StringComparison.OrdinalIgnoreCase, out values))
+                {
+                    values = new JArray();
+                    recognizerResult.Entities[entityResult.Type] = values;
+                }
+
+                // The Entity type names are not consistent, map everything to camelcase so we can process them cleaner.
+                var entity = JObject.FromObject(entityResult);
+                ((JArray)values).Add(entity.GetValue("text", StringComparison.InvariantCulture));
+
+                // get/create $instance
+                JToken instanceRoot;
+                if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                {
+                    instanceRoot = new JObject();
+                    recognizerResult.Entities["$instance"] = instanceRoot;
+                }
+
+                // add instanceData
+                JToken instanceData;
+                if (!((JObject)instanceRoot).TryGetValue(entityResult.Type, StringComparison.OrdinalIgnoreCase, out instanceData))
```

Comment:
can you declare JToken inline by the out

Reply:
Fixed.